ops/jujuversion.py

```diff
@@ -46,18 +58,8 @@ def __lt__(self, other):
         elif not isinstance(other, JujuVersion):
             raise RuntimeError(f'cannot compare Juju version "{self}" with "{other}"')
 
-        if self.major != other.major:
-            return self.major < other.major
-        elif self.minor != other.minor:
-            return self.minor < other.minor
-        elif self.tag != other.tag:
-            if not self.tag:
-                return False
-            elif not other.tag:
-                return True
-            return self.tag < other.tag
-        elif self.patch != other.patch:
-            return self.patch < other.patch
-        elif self.build != other.build:
-            return self.build < other.build
+        for prop in ['major', 'minor', 'tag', 'patch', 'build']:
+            lt = self._less_than_property(other, prop)
+            if lt is not None:
+                return lt
```

Comment:
The original version had 14 lines.. the new version has 14 lines, except it jumps across two different functions back and forth repeatedly, uses more indirection, and is harder to read. It's also a bit of a lie as it pretends to be general until it's not, by special casing the handling of "tag".

I'd recommend preserving the existing function.